chain/network-primitives/src/network_protocol/edge.rs

```diff
@@ -217,6 +224,15 @@ impl Edge {
             None
         }
     }
+
+    // Checks if edge was created before a given timestamp.
+    pub fn is_edge_older_than(&self, utc_timestamp: Utc) -> bool {
+        if self.nonce() < EDGE_MIN_TIMESTAMP_NONCE {
+            // Old-style nonce - for now, assume that they are always fresh.
```

Comment:
how about using a highest bit for distinguishing these instead (0 -old, 1 - new), so that there is no artificial timestamp boundary?

Reply:
that's a nice idea, but I'm worried about two things:
- it would make it a little bit harder to debug (unless we adjust all the logs)
- might be harder for future updates (as we would already have first bit always occupied)

If you strongly believe in this approach - we can chat offline.

Reply:
ack, this is fine too